### DISCUSSION TITLE
ShowDevTools with SetAsPopup no longer creates a new window

### DISCUSSION THREAD

**Discussion by oscar-articulate** [cefsharp/CefSharp#5004] [General]
### Is there an existing issue for this?

- [X] I have searched both open/closed issues, no issue already exists.

### CefSharp Version

130.1.90

### Operating System

Windows 10

### Architecture

x64

### .Net Version

net4.8

### Implementation

WinForms

### Reproduction Steps

When using `124.3.80`, the below code would create a new windows form with our application (the host's) icon to host the dev tools in it. It would also show up in the task bar.

Now, with `130.1.90`, the below code creates an edgeless form (?) with no title bar, close/minimize/maximize, etc.

```
var host = browser.GetBrowserHost();

var windowInfo = new WindowInfo();
windowInfo.SetAsPopup(host.GetWindowHandle(), "DevTools");

browser.ShowDevTools(windowInfo);
```

### Expected behavior

a new window (forms/wpf) with title bar, minimize/maximize/close, etc, is created that hosts the dev tools, and inherits the main application's icon. 

![image](https://github.com/user-attachments/assets/d4de7a6d-4036-4f20-96f9-d40a11efed47)

### Actual behavior

an edgeless dev tools shows up with no title bar, minimize/maximize/close is created, and a (blue) chromium icon is shown
![image](https://github.com/user-attachments/assets/4d91181c-8051-44f9-9e6d-8738e42c8aa3)

### Regression?

yes, worked in version 124.3.80

### Known Workarounds

My workaround is to create a new form and use the extensions:

```
var devToolsForm = new Form(); 
var devToolsControl = CefSharp.WinForms.WebBrowserExtensions.ShowDevToolsDocked(this, devToolsForm);
devToolsForm.Show(mainApplicationForm);
```

and another workaround is:

```

            var windowInfo = new WindowInfo
            {
                WindowlessRenderingEnabled = true
            };

```

This applies to both WPF and Forms.

I know that the alloy stuff was removed, and I'm guessing it's related to this.

### Does this problem also occur in the CEF Sample Application
yes; modify in BrowserForm.cs

```
        private async void ShowDevToolsMenuItemClick(object sender, EventArgs e)
        {
            var control = GetCurrentTabControl();
            if (control != null)
            {
                var isDevToolsOpen = await control.CheckIfDevToolsIsOpenAsync();
                if (!isDevToolsOpen)
                {
                    var host = control.Browser.GetBrowserHost();

                    var windowInfo = new WindowInfo();
                    windowInfo.SetAsPopup(host.GetWindowHandle(), "DevTools");

                    control.Browser.ShowDevTools(windowInfo);
                }
            }
        }
```

**Comment by amaitland** [comment]
Converting to discussion as this is an `upstream` change in behavior, `CefSharp` just wraps the `API`.

Calling `ShowDevTools()` without any args in version `131.3.10` looks fine to me.

![image](https://github.com/user-attachments/assets/00775f78-ff27-4ce8-90db-a48a30e313f4)

> I know that the alloy stuff was removed, and I'm guessing it's related to this.

Very likely. Remember that `CEF` has it's own issue tracker at https://github.com/chromiumembedded/cef/issues

If there's a specific use case that worked previously then you'd need to open an issue there (post a link back here for reference if you do).

**Reply by oscar-articulate** [reply]
@amaitland  Yes if you just call `ShowDevTools()` without any args it shows up fine.
The bug/regression seems to be if you use `SetAsPopup`:

```
windowInfo.SetAsPopup(host.GetWindowHandle(), "DevTools");
```

**Reply by amaitland** [reply]
That would need to be fixed in `CEF`.

https://github.com/chromiumembedded/cef/issues
